src/service/data_products/taxa_count.py

```diff
@@ -234,9 +243,43 @@ async def get_taxa_counts(
         await _add_subset_data_in_place(q, store, collection_id, load_ver, rank, dp_proc)
         # For now always sort by the std data. See if ppl want sort by match/selection data
         # before implementing something more sophisticated.
+    _sort_taxa_counts(q)
+
     return _taxa_counts(dp_match=dp_match, dp_sel=dp_sel, data=q)
 
 
+def _sort_dict_list(
+        dict_list: list[dict],
+        key: str,
+        reverse: bool = True):
+    # Sort taxa count records, a list of dicts, in place by the key of dictionary with None values last.
+    # Key values should be either numeric or None.
+    # Certain keys, such as 'sel_count' or 'match_count,' may not exist in all dictionaries.
+    # For instance, these keys might be absent if the corresponding match or selection process did not occur/complete.
+    # In such cases, the default value of float('-inf') is used to ensure that the dictionaries are sorted correctly.
+    dict_list.sort(key=lambda x: (int(x.get(key)) if x.get(key) is not None else float('-inf')), reverse=reverse)
```

Comment:
Why is the int conversion necessary?

Reply:
I just want to catch in case the field value cannot be casted to int which should not happen. But just in case.

Reply:
Alright, can you add a comment explaining why it's there?

Reply:
Wait, hold on - do you want it to throw an error? Because floats will just get silently truncated which seems wrong

Reply:
👍 it makes `_sort_dict_list` unnecessarily complicated. Given our use case, it's probably not worth it.

Reply:
Sorry, what are you saying isn't worth it?

Reply:
And should floats be allowed here since these are counts?

Reply:
updated to  `x[k]` since key should always exist now.

Reply:
: